CodeGen/Generators/QuantityJsonFilesParser.cs

```diff
@@ -124,5 +126,246 @@ private static Localization[] GetLocalizationForPrefixUnit(IEnumerable<Localizat
+                }
+            }
+
+            if (oldUnits.J is { } oldLuminosity && dimensions.J == degree)
+            {
+                if (TryPrefixUnit(oldLuminosity, degree, prefix, out var newLuminosity))
+                {
+                    baseUnits.J = newLuminosity;
+                    return true;
+                }
+            }
+
+            if (oldUnits.M is {} oldMass && dimensions.M == degree)
+            {
+                if (TryPrefixUnit(oldMass, degree, prefix, out var newMass))
+                {
+                    baseUnits.M = newMass;
+                    return true;
+                }
+            }
+
+            if (oldUnits.Θ is {} oldTemperature && dimensions.Θ == degree)
+            {
+                if (TryPrefixUnit(oldTemperature, degree, prefix, out var newTemperature))
+                {
+                    baseUnits.Θ = newTemperature;
+                    return true;
+                }
+            }
+            
+            if (oldUnits.T is {} oldTime && dimensions.T == degree)
+            {
+                if (TryPrefixUnit(oldTime, degree, prefix, out var newTime))
+                {
+                    baseUnits.T = newTime;
+                    return true;
+                }
+            }
+
+            return false;
+        }
+
+        private static bool TryPrefixUnit(string oldBase, int degree, Prefix prefix, [NotNullWhen(true)] out string? newBaseUnit)
+        {
+            if (PrefixedStringFactors.TryGetValue(oldBase, out (string, int) existingPrefixMapping) && PrefixFactors.TryGetValue(prefix, out var prefixFactor))
+            {
+                var baseUnit = existingPrefixMapping.Item1;
+                var currentFactor = existingPrefixMapping.Item2;
+                var (quotient, remainder) = int.DivRem(prefixFactor, degree);
+                if (remainder == 0 && PrefixFactorsByValue.TryGetValue(currentFactor + quotient, out Prefix calculatedPrefix))
+                {
+                    if (BaseUnitPrefixConversions.TryGetValue((baseUnit, calculatedPrefix), out newBaseUnit))
+                    {
+                        return true;
+                    }
+                }
+            }
+
+            newBaseUnit = null;
+            return false;
+        }
+        
+        /// <summary>
+        /// A dictionary that maps metric prefixes to their corresponding exponent values.
+        /// </summary>
+        /// <remarks>
+        /// This dictionary excludes binary prefixes such as Kibi, Mebi, Gibi, Tebi, Pebi, and Exbi.
+        /// </remarks>
+        private static readonly Dictionary<Prefix, int> PrefixFactors = PrefixInfo.Entries
+            .Where(x => x.Key is not (Prefix.Kibi or Prefix.Mebi or Prefix.Gibi or Prefix.Tebi or Prefix.Pebi or Prefix.Exbi)).ToDictionary(pair => pair.Key,
+                pair => (int)Math.Log10(double.Parse(pair.Value.Factor.TrimEnd('d'), NumberStyles.Any, CultureInfo.InvariantCulture)));
```

Comment:
I was a little worried when casting to `(int)` but I've checked the results, there are no rounding errors..
Still, we could simply expose these as properties on the `Prefix` type.

Reply:
> I was a little worried when casting to `(int)` but I've checked the results, there are no rounding errors.. Still, we could simply expose these as properties on the `Prefix` type.

Although these might have to be two properties (given that we have both binary and decimal prefixes)..